lib/vector-buffers/src/topology/channel/poll_semaphore.rs

```diff
@@ -0,0 +1,118 @@
+use futures::ready;
```

Comment:
This is here because while `tokio_util` does already have `PollSemaphore`, it does not have a method to acquire multiple permits in a single call, and the process to upstream that just isn't fast enough to have done it before submitting this PR.

Similar to `PollNotify`, this can go away once we are able to excise the `Sink` code from the buffer topology code.

Reply:
Is it worth _also_ upstreaming it as well? Or no because of the planned removal eventually

Reply:
I will likely submit a PR upstream for it, but it's questionable whether or not we'll get to remove this from the code and switch to an upstreamed version before we remove the _need_ for it.